larpandoracontent/LArReclustering/ThreeDReclusteringAlgorithm.cc

```diff
@@ -0,0 +1,483 @@
+/**
+ *  @file   larpandoracontent/LArReclustering/ThreeDReclusteringAlgorithm.cc
+ *
+ *  @brief  Implementation file for the reclustering algorithm class.
+ *
+ *  $Log: $
+ */
+
+#include "Pandora/AlgorithmHeaders.h"
+#include "Managers/ClusterManager.h"
+
+#include "larpandoracontent/LArHelpers/LArPfoHelper.h"
+#include "larpandoracontent/LArHelpers/LArClusterHelper.h"
+#include "larpandoracontent/LArHelpers/LArPcaHelper.h"
+#include "larpandoracontent/LArHelpers/LArMCParticleHelper.h"
+
+#include "larpandoracontent/LArReclustering/ThreeDReclusteringAlgorithm.h"
+
+using namespace pandora;
+
+namespace lar_content
+{
+
+ThreeDReclusteringAlgorithm::ThreeDReclusteringAlgorithm():
+    m_pfoListName("ShowerParticles3D"),
+    m_hitThresholdForNewPfo(0),
+    m_fomThresholdForReclustering(0.3)
+{
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+ThreeDReclusteringAlgorithm::~ThreeDReclusteringAlgorithm()
+{
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+StatusCode ThreeDReclusteringAlgorithm::Run()
+{
+    //Only proceed if successfuly get shower pfo list, and it has at least one pfo
+    const PfoList *pShowerPfoList(nullptr);
+    if (!((STATUS_CODE_SUCCESS == PandoraContentApi::GetList(*this, m_pfoListName, pShowerPfoList)) && pShowerPfoList->size()))
+    {
+        return STATUS_CODE_SUCCESS;
+    }
+
+    m_PfosForReclusteringListName = "newShowerParticles3D";
```

Comment:
Should this be a configurable?

Reply:
I think not, as this list is purely internal to this algorithm.